src/Utilities/Functional.hpp

```diff
@@ -0,0 +1,136 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Higher order function objects similar to `std::plus`, etc.
```

Comment:
@moxcodes would you be able to expand with some examples in this documentation when you do the other math ones?

Reply:
Yep, will do.

Reply:
Thank you!